src/testdrive/src/parser.rs

```diff
@@ -122,7 +122,15 @@ fn parse_builtin(line_reader: &mut LineReader) -> Result<BuiltinCommand, InputEr
                 pos,
             });
         }
-        args.insert(pieces[0].to_owned(), pieces[1].to_owned());
+        if let Some(original) = args.insert(pieces[0].to_owned(), pieces[1].to_owned()) {
+            return Err(InputError {
+                msg: format!(
+                    "duplicate argument '{}' first value: {} second: {}",
```

Comment:
Could you update this to use proper grammar? As written it's going to read `duplicate argument 'foo' first value: {"foo": "bar"} second: {"bar": "foo"}` which seems hard to parse. I think just `argument '{}' specified twice` might be more clear. testdrive commands tend to be short so I'm not sure it enhances readability to mention the values.